### PR TITLE
Disallow instantiation with non dynamic matrices.

### PR DESCRIPTION
**Pull request by simonlynen**
Add the suggestion of @HannesSommer to disallow instantiation of non dynamic matrices. Also adds tests for this.

@HannesSommer @tcies ptal

### CODE REVIEW

**Comment by HannesSommer** [comment]
Awesome! :+1: for including compiler failure tests!

File: nabo/nabo.h [hunk 1]
```diff
@@ -365,6 +365,44 @@ namespace Nabo
 		 * 	\return an object on which to run nearest neighbour queries */
 		static NearestNeighbourSearch* createKDTreeTreeHeap(const Matrix& cloud, const Index dim = std::numeric_limits<Index>::max(), const unsigned creationOptionFlags = 0, const Parameters& additionalParameters = Parameters());
 		
+
+
+    //! Prevent creation of trees with the wrong matrix type. Currently only dynamic size matrices are supported.
```
Comment (HannesSommer) [inline nabo/nabo.h]: I'm afraid, you've got the wrong indentation white space.. This project is tab indented.

File: tests/CMakeLists.txt [hunk 2]
```diff
@@ -79,4 +79,35 @@ add_executable(knnepsilon knnepsilon.cpp)
 target_link_libraries(knnepsilon ${LIB_NAME} ${EXTRA_LIBS} ${Boost_LIBRARIES})
 
 add_executable(knnbucketsize knnbucketsize.cpp)
-target_link_libraries(knnbucketsize ${LIB_NAME} ${EXTRA_LIBS} ${Boost_LIBRARIES})
\ No newline at end of file
+target_link_libraries(knnbucketsize ${LIB_NAME} ${EXTRA_LIBS} ${Boost_LIBRARIES})
+
+# Ensure that users cannot instantiate a tree with wrong matrix types.
+macro(try_compile_disallowed_types MAT_TYPE TREE_TYPE EXPECT)
```
Comment (HannesSommer) [inline tests/CMakeLists.txt]: The ```EXPECT``` variable is in conflict with ```disallowed``` here somehow, I believe. Maybe something ```try_compile_cloud_types``` would be more accurate?

File: tests/CMakeLists.txt [hunk 3]
```diff
@@ -79,4 +79,35 @@ add_executable(knnepsilon knnepsilon.cpp)
...
+  try_compile(COMPILE_SUCCEEDED
+                "${CMAKE_CURRENT_BINARY_DIR}"
+                "${CMAKE_CURRENT_SOURCE_DIR}/invalid_matrix_types_disallowed.cpp"
```
Comment (HannesSommer) [inline tests/CMakeLists.txt]: Same here with the ```disallowed```, I guess.

File: tests/CMakeLists.txt [hunk 4]
```diff
@@ -79,4 +79,35 @@ add_executable(knnepsilon knnepsilon.cpp)
...
+                COMPILE_DEFINITIONS "-D${MAT_TYPE} -D${TREE_TYPE} -I${CMAKE_CURRENT_SOURCE_DIR}/../ -I${EIGEN_INCLUDE_DIR} -l${LIB_NAME}"
+                CMAKE_FLAGS "-DCMAKE_CXX_LINK_EXECUTABLE='echo not linking now...'"
+                OUTPUT_VARIABLE OUTPUT)
+
+  if (${EXPECT} STREQUAL "TRUE") 
+    if(NOT COMPILE_SUCCEEDED)
+      message( FATAL_ERROR "Compiling a test executable failed." )
```
Comment (HannesSommer) [inline tests/CMakeLists.txt]: Will this be detected by Jenkins as an error? I would assume so, but I'm not sure.

**Comment by HannesSommer** [comment]
If you don't have the time for corrections you agree with: I could fill in.

**Comment by simonlynen** [comment]
Thanks for the review, Ill fix these tomorrow.

File: tests/CMakeLists.txt (hunk 4 above)
Comment (simonlynen) [inline tests/CMakeLists.txt]: Yes, cmake errors are in the logparser rules

File: nabo/nabo.h (hunk 1 above)
Comment (simonlynen) [inline nabo/nabo.h]: Yes :)

**Comment by tcies** [comment]
Thanks @simonlynen !

**Pull request merged**